### PR TITLE
Add a way to handle dynamic options

### PR DESCRIPTION
**Pull request by LupusVII**
Hi!
Second and last PR from me ;)

For a project, I worked with a framework and a lot of classes could be useful. But since it could evolve, it would be too heavy to update the CSS and the Style Manager. So I made a hook who can be triggered when the options of the widget are loaded.
In the function, I could open the CSS, parse the classes and return them correctly formatted.
I used a keyword to know I'm in the right widget too.

Let me know if it's something you might add in your repo :)
Thanks again!

### CODE REVIEW

**Comment by doishub** [comment]
Hi, thanks for contributing.

Basically, we can gladly implement a hook that gives you the ability to dynamically populate select box options.

I think the position of the hook is not quite in the right place yet. Here I would suggest to place the hook below the `foreach` to not only manipulate the CSS classes available in the system but also to be able to add some dynamically. In your function you would have to go through the whole `$options` array again or you could add new values to it.

File: src/Resources/contao/widgets/ComponentStyleSelect.php [hunk 1]
```diff
@@ -138,10 +138,24 @@ public function generate()
             $opts = StringUtil::deserialize($objStyleGroups->cssClasses);
 
             foreach ($opts as $opt) {
-                $arrFieldOptions[] = array(
+                $option = array(
                     'label' => $opt['value'] ?: $opt['key'],
                     'value' => $opt['key']
                 );
+
+                // skip third-party fields
```
Comment (doishub) [inline src/Resources/contao/widgets/ComponentStyleSelect.php:146]: Change description to: `Dynamically change or expand group options`

File: src/Resources/contao/widgets/ComponentStyleSelect.php [hunk 2]
```diff
@@ -138,10 +138,24 @@ public function generate()
...
+                if (isset($GLOBALS['TL_HOOKS']['styleManagerGetDynamicFieldOptions']) && \is_array($GLOBALS['TL_HOOKS']['styleManagerGetDynamicFieldOptions']))
```
Comment (doishub) [inline src/Resources/contao/widgets/ComponentStyleSelect.php:147]: The names of the hooks in the StyleManager always tell the current position of the context. Here we should rather name the hook like this: `styleManagerGroupFieldOptions`.

File: src/Resources/contao/widgets/ComponentStyleSelect.php (hunk 1 above)
Comment (LupusVII) [reply src/Resources/contao/widgets/ComponentStyleSelect.php:146]: Done in b6e98db

File: src/Resources/contao/widgets/ComponentStyleSelect.php (hunk 2 above)
Comment (LupusVII) [reply src/Resources/contao/widgets/ComponentStyleSelect.php]: Done in 57a7ab1

File: src/Resources/contao/widgets/ComponentStyleSelect.php (hunk 1 above)
Comment (doishub) [reply src/Resources/contao/widgets/ComponentStyleSelect.php]: You have unfortunately adjusted the wrong line. 🙈

**Comment by doishub** [comment]
I have made the rest of the adjustments discussed. Could you please check if it works for you. Then I would merge the PR.

**Comment by doishub** [comment]
The solution shown here will be considered in the next release. Responsible branch of the next version can be viewed [here](https://github.com/oveleon/contao-component-style-manager/tree/feature/bundle-config).

**Pull request closed without merging**